common/src/api/external/mod.rs

```diff
@@ -472,6 +472,7 @@ pub enum ResourceType {
     Dataset,
     Disk,
     DiskAttachment,
+    DownloadArtifact,
```

Comment:
It looks like we're adding / have added a bunch of stuff here that's only appropriate to the internal API.  I'm guessing that's because of the way we try to construct `external::Error`s from diesel errors using the helper function that takes a resource type.  Maybe we should have a different helper that uses a different enum for internal stuff?  Doesn't have to be done here.

Reply:
Agreed. Filed https://github.com/oxidecomputer/omicron/issues/532 to track.